### DISCUSSION TITLE
How to build myself environment?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [thu-ml/tianshou#804] [Q&A]
I want to implement reinforcement learning with `tianshou`, but I need to build myself environment.

My enviroment is a 3D maze (`128x128x128 matrix`), and the network should be a 3D convolution neural network. 

How can I build myself enviroment? Should I subclass `BaseVectorEnv`? Or is there any example to build myself environment?

Thanks for any suggestion~~~

**Comment by Trinkle23897** [comment]
https://tianshou.readthedocs.io/en/master/tutorials/cheatsheet.html#user-defined-environment-and-different-state-representation

**Reply by konnovdev** [reply]
note that the link to this doc has changed now, here's the updated link: https://tianshou.org/en/stable/01_tutorials/07_cheatsheet.html#user-defined-environment-and-different-state-representation
